File: OC.Domain/Models/Accounting/Pledge.cs
using OC.Domain.Interfaces;
using OC.Domain.Models.Branches;
using OC.Domain.Models.Members;

namespace OC.Domain.Models.Accounting
{
    public class Pledge : AuditableEntity, IEntityBase
    {
        public Pledge() { }

        public long Id { get; set; }
        public long? MemberId { get; set; }
        public long? MinistryId { get; set; }
        public long? ChurchProgramSessionId { get; set; }
        public string? Description { get; set; }
        public double Amount { get; set; }
        public double PaidAmount { get; set; } = 0;
        public DateTime? PaymentDate { get; set; }
        public DateTime? PaidDate { get; set; }
        public bool? IsRecurring { get; set; } = false;
        public int? RecurringReminderDay { get; set; } = 1;
        // TODO: INCLUDE LOGIC FOR INVOICING AND TRACKING OF PAYMENTS

        // Relationships
        public virtual Member? Member { get; set; }
        public virtual Ministry? Ministry { get; set; }
        public virtual ChurchProgramSession? ChurchProgramSession { get; set; }
    }
}

File: OC.UI/src/app/blci/api/accounting/pledge.ts
import { Ministry } from "../branches/ministry";
import { Member } from "../members/member";


export interface Pledge {
    id: number;
    memberId: number | null;
    ministryId: number | null;
    description: string | null;
    amount: number;
    paidAmount: number;
    paymentDate: string | null;
    paidDate: string | null;
    member: Member;
    ministry: Ministry;
}
